src/types/index.ts

```diff
@@ -592,6 +592,22 @@ export class StarknetContract {
         return this.adaptOutput(functionName, executed.stdout.toString());
     }
 
+    /**
+     * Computes message fee estimation
+     * @param {string} fromAddress - Address of message source
+     * @param {string} functionName - Function name for entry point selector
+     * @param {Array<Numeric>} payload - Payload to send
+     * @returns Fee estimation
+     */
+    async estimateMessageFee(fromAddress: string, functionName: string, payload: Numeric[]) {
```

Comment:
The user shouldn't be forced to pass in a numeric array. They could provide an object with parameters corresponding to function argument names, which we then check and convert to an array using `adaptInput` (so - as with `contract.call`).

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/312/commits/640e641525992f63abc27f7cc87f1342a68ccd0d